File: tests/fixtures/supabase/supabase-realtime-missing-filter-fixed/02-non-postgres-changes-adversarial.ts
import { supabase } from '../client.js';

export function subscribePresence(userId: string, onStatus: (payload: unknown) => void) {
  return supabase
    .channel('presence')
    .on('presence', { event: 'sync' }, () => onStatus(null))
    .subscribe();
}

File: tests/fixtures/supabase/supabase-realtime-missing-filter-fixed/02_non_postgres_changes_adversarial.py
from lib.client import supabase


def handle_broadcast(payload):
    print(payload)


def subscribe_broadcast():
    channel = supabase.channel("room1")
    # on_broadcast is a different event type entirely — should not be
    # confused with the postgres_changes filter requirement.
    return channel.on_broadcast(event="cursor-pos", callback=handle_broadcast).subscribe()
